genie-web/src/main/java/com/netflix/genie/web/aspect/HealthCheckMetricsAspect.java

```diff
@@ -0,0 +1,248 @@
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.aspect;
+
+import com.google.common.collect.Maps;
+import com.netflix.genie.core.util.MetricsConstants;
+import com.netflix.genie.core.util.MetricsUtils;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Aspect woven into Spring Boot 'Health' machinery to publish metrics such as time taken, errors and other signals
+ * useful for dashboards and alerting.
+ *
+ * @author mprimi
+ * @since 3.2.4
+ */
+@Aspect
+@Component
+@Slf4j
+public class HealthCheckMetricsAspect {
+
+    private static final String HEALTH_ENDPOINT_TIMER_NAME = "genie.health.endpoint.timer";
+    private static final String HEALTH_INDICATOR_TIMER_METRIC_NAME = "genie.health.indicator.timer";
+    private static final String HEALTH_INDICATOR_COUNTER_METRIC_NAME = "genie.health.indicator.counter";
+    private static final String HEALTH_FAILURES_COUNTER_METRIC_NAME = "genie.health.failure.counter";
+    private static final String HEALTH_INDICATOR_CLASS_TAG_NAME = "healthIndicatorClass";
+    private static final String HEALTH_INDICATOR_NAME_TAG_NAME = "healthIndicatorName";
+
+    private final Registry registry;
+    private final Id healthEndpointTimerId;
+    private final Id healthIndicatorTimerId;
+    private final Id healthIndicatorCounterId;
+    private final Id healthIndicatorFailureCounterId;
+
+    /**
+     * Autowired constructor.
+     * @param registry metrics registry
+     */
+    @Autowired
+    public HealthCheckMetricsAspect(final Registry registry) {
+        this.registry = registry;
+        this.healthEndpointTimerId = registry.createId(HEALTH_ENDPOINT_TIMER_NAME);
+        this.healthIndicatorTimerId = registry.createId(HEALTH_INDICATOR_TIMER_METRIC_NAME);
+        this.healthIndicatorCounterId = registry.createId(HEALTH_INDICATOR_COUNTER_METRIC_NAME);
```

Comment:
Timers automatically include a counter metric. Creating a separate one is redundant.